### PR TITLE
feat(timm_resnest): add timm ResNeSt image-classification family

### PR DESCRIPTION
**Pull request by zhenshanx-nv**
## Background

TensorRT-Model-Connect has no ResNeSt family. ResNeSt is listed by
wang-xinyu/tensorrtx and cannot reuse `timm_resnet` or `timm_seresnet`: its
bottleneck replaces the 3x3 convolution with split attention, which is a
different gate shape from squeeze-and-excitation, and the `d` variants also
change where the stride lands.

## Exit Criteria

- `timm/resnest50d.in1k` builds and classifies through the
  `timm_resnest_image_classification` runtime strategy.
- Logits match the timm reference on the same weights and the same input.
- The family is registered in the shared matrices, workloads, perf release, and
  website tables, and the existing static encapsulation tests still pass.

Non-goals: quantized builds and tensor-parallel builds both raise
`NotImplementedError`; only `resnest50d` is registered, and only the `d`
variants' stride placement is implemented.

## Implementation

New `timm_resnest` family, following the layout the other timm families use: a
Python builder plugin that emits the network through the TensorRT Network API,
a C++ runtime model with its own image-preprocess seam, and an e2e model
directory with one manifest.

`_split_attention` is the part worth reading. The convolution emits `radix`
copies of the output channels; the gate is computed from their sum and
normalised with a **softmax across the radix axis**, so the copies compete for
each channel. The SE families this repo already has use a per-channel sigmoid,
where each channel is gated independently — substituting one for the other
gives a plausible wrong answer rather than an error, so the distinction is
commented in place.

Stage depths and the deep stem come from the checkpoint. The stem convolutions
are picked out by tensor rank rather than by index, because the norms
interleaved in the same `nn.Sequential` also have a `.weight`.

Two things the checkpoint cannot record are stated as constants with a comment
saying why, both specific to the `d` variants: a stride never lands on the
split-attention convolution but on an average pool placed after it, and the
shortcut is average-pooled before its 1x1 projection.

### Change categories

- [x] Model or runtime behavior
- [ ] Public API
- [ ] ABI
- [ ] Bundle or artifact format
- [ ] Dependencies
- [x] Documentation only
- [ ] CI or developer tooling

### Commands and Results

Unit and tooling suite:

```
python -m pytest tests/builder/ tests/tools/ tests/e2e_harness/ -q -n 4 \
  --dist=worksteal --import-mode=importlib -p no:cacheprovider --tb=line
3 failed, 4042 passed, 8 skipped, 23 warnings in 265.92s
```

The 3 failures are pre-existing on this machine and unrelated to this change:
two `test_model_proof_runner.py` cases call `/bin/cp --reflink` on a filesystem
that does not support cloning, and `TestAddGeluFcMlp::test_silu_activation`
fails the same way on an unmodified checkout.

Numerical parity, TRT fp32 against timm on identical weights and one fixed
random 1x3x224x224 input:

```
timm reference: missing=0 unexpected=0
max|diff| = 0.006126
corr      = 0.99999914
argmax    TRT=21 timm=21
top5 overlap = 5/5
RESULT: PASS
```

The weight-name check reports `missing=0 unexpected=0`, so every tensor in the
checkpoint is consumed by the builder and nothing is invented.

### Hardware, Environment, and Revisions

| Item | Value |
| --- | --- |
| Repository head | `92db1113` |
| Checkpoint | `timm/resnest50d.in1k` @ `2239b0d822126dee3bc7a5357514de35b8f42092` |
| Reference | timm 1.0.28 |
| GPU | NVIDIA A100-SXM4-80GB |
| Container | project `Dockerfile.dev.x86`, Ubuntu 24.04, Python 3.12 |
| Precision | parity run fp32; manifest builds fp16 |

### Not Run / Remaining Gaps

- The full e2e harness case was not run locally; it needs the CI image cache.
  It is covered by the internal gate.
- Only fp32 was compared against the reference. The fp16 path is exercised by
  the manifest but not separately compared.
- Only `resnest50d` was tested. A non-`d` ResNeSt would need the stride moved
  back onto the convolution, which this builder does not do.
- Only `cardinality == 1` was exercised. The grouped path in
  `_split_attention` is written generally but no registered checkpoint uses it.

## Notes For Future Readers

Suggested review order: `_split_attention` in `plugin.py`, then `_block`; the
registration diffs are mechanical and mirror the previous timm families.

If a future ResNeSt is added and lands close but wrong, check the two stride
constants first. They are architecture facts that no checkpoint records, and
getting either wrong still produces correctly shaped tensors throughout.

### Risk level

- [x] Low
- [ ] Medium
- [ ] High

The change is additive. It creates a new family directory and inserts adjacent
lines in the shared registration files; no existing family shares code with it.

### CODE REVIEW

**Comment by zhenshanx-nv** [comment]
Superseded by #1215, which implements this family in the post-cutover layout introduced by #1188. This branch targets `python/tensorrt_model_connect/families/`, which no longer exists, so it cannot be rebased.
